test_conformance/SVM/test_enqueue_api.cpp

```diff
@@ -113,115 +113,166 @@ int test_svm_enqueue_api(cl_device_id deviceID, cl_context c, cl_command_queue q
     sizeof(cl_ulong16),
   };
 
-  for (size_t i = 0; i < ( sizeof(typeSizes) / sizeof(typeSizes[0]) ); ++i)
-  {
-    //generate initial data
-    std::vector<cl_uchar> fillData0(typeSizes[i]), fillData1(typeSizes[i], 0), fillData2(typeSizes[i]);
-    generate_data(fillData0, typeSizes[i], seed);
-    generate_data(fillData2, typeSizes[i], seed);
+  enum allocationTypes {
+    host,
+    svm
+  };
 
-    cl_uchar *srcBuffer = (cl_uchar *)clSVMAlloc(context, CL_MEM_READ_WRITE, elementNum * typeSizes[i], 0);
-    cl_uchar *dstBuffer = (cl_uchar *)clSVMAlloc(context, CL_MEM_READ_WRITE, elementNum * typeSizes[i], 0);
+  typedef struct _TestType {
+    allocationTypes srcAlloc;
+    allocationTypes dstAlloc;
+    _TestType(allocationTypes type1, allocationTypes type2): srcAlloc(type1), dstAlloc(type2){}
+  } TestType;
 
-    clEventWrapper userEvent = clCreateUserEvent(context, &error);
-    test_error(error, "clCreateUserEvent failed");
+ typedef std::vector<TestType> TestTypes;
 
-    clEventWrapper eventMemFill;
-    error = clEnqueueSVMMemFill(queue, srcBuffer, &fillData0[0], typeSizes[i], elementNum * typeSizes[i], 1, &userEvent, &eventMemFill);
-    test_error(error, "clEnqueueSVMMemFill failed");
+ TestTypes testTypes;
+ testTypes.push_back(TestType(host, svm));
+ testTypes.push_back(TestType(svm, host));
+ testTypes.push_back(TestType(svm, svm));
```

Comment:
What about host/host? We might as well throw it in and have exhaustive coverage.

Reply:
I discussed this with @bashbaug . We think this case is not very useful and probably spec clarification is needed to turn off such case.

Reply:
Hmm, I'd argue that it is clearly allowed by the spec and has been since 2.0. I don't think we can change this now and though I agree it doesn't seem very useful I think we need to cover it here.